plugins/gradle/gradle-plugin/src/main/java/org/wildfly/swarm/plugin/gradle/StartTask.java

```diff
@@ -0,0 +1,354 @@
+package org.wildfly.swarm.plugin.gradle;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.gradle.BuildListener;
+import org.gradle.BuildResult;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.artifacts.repositories.MavenArtifactRepository;
+import org.gradle.api.initialization.Settings;
+import org.gradle.api.invocation.Gradle;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.bundling.Jar;
+import org.gradle.api.tasks.bundling.War;
+import org.gradle.initialization.BuildCancellationToken;
+import org.wildfly.swarm.bootstrap.util.BootstrapProperties;
+import org.wildfly.swarm.fractions.FractionDescriptor;
+import org.wildfly.swarm.fractions.FractionList;
+import org.wildfly.swarm.fractions.FractionUsageAnalyzer;
+import org.wildfly.swarm.tools.ArtifactResolvingHelper;
+import org.wildfly.swarm.tools.ArtifactSpec;
+import org.wildfly.swarm.tools.BuildTool;
+import org.wildfly.swarm.tools.DependencyManager;
+import org.wildfly.swarm.tools.exec.SwarmExecutor;
+import org.wildfly.swarm.tools.exec.SwarmProcess;
+
+public class StartTask extends DefaultTask {
+
+    private static final String EXCLUDE_PREFIX = "!";
+    private static final String INDENTATION_DELIMITER = "\n  ";
+
+    private ThorntailExtension extension;
+    private boolean waitForProcess = false;
+
+    public StartTask() {
+        extension = getProject().getExtensions().getByType(ThorntailExtension.class);
+    }
+
+    void waitForProcess() {
+        waitForProcess = true;
+    }
+
+    @TaskAction
+    public void startApplication() {
+        final SwarmExecutor executor;
+        if (extension.isUseUberJar()) {
```

Comment:
Is there a reason why you wouldn't want to use the `UberJar` only? Meaning, `thorntail-start` just blindly depends on `thorntail-package`. That way, we don't need to incur the maintenance of the build tooling. Did you run in to any issues specifically?

Reply:
I think it could shave couple of seconds off the build time, if you don't need to build the UberJar and use the compiled classes instead.

Reply:
This was also consistency-with-maven-plugin thing. I'm not absolutely sure about the practical aspects, only that I would personally appreciate an option of not having to build an uber-jar.